extensions/wikia/WikiFactory/DeadWikis/maintenance.php

```diff
@@ -382,6 +398,74 @@ private function disableDiscussion( $cityId ) {
 		}
 	}
 
+	// Gets the most recent post from the specified site and returns whether it was made less than 60 days ago
+	private function isActiveSite($site_id)
+	{
+		$apiClient = $this->getSitesApi()->getApiClient();
+
+		$resourcePathTemplate = "/{siteId}/posts";
+		$httpBody = '';
+		$queryParams = array();
+		$headerParams = array();
+
+		// header params
+		$_header_accept = $apiClient->selectHeaderAccept(array('application/hal+json'));
+		if (!is_null($_header_accept)) {
+			$headerParams['Accept'] = $_header_accept;
+		}
+		$headerParams['Content-Type'] = $apiClient->selectHeaderContentType(array('application/json'));
+
+		// path params
+		$resourcePath = str_replace(
+				"{siteId}",
+				$apiClient->getSerializer()->toPathValue($site_id),
+				$resourcePathTemplate
+		);
+
+		// only need most recent post
+		$queryParams['limit'] = 1;
+
+
+		$response = null;
+
+		// make the API Call
+		try {
+			list($rawResponse, $statusCode, $httpHeader) = $apiClient->callApi(
+					$resourcePath,
+					'GET',
+					$queryParams,
+					$httpBody,
+					$headerParams,
+					'object',
+					$resourcePathTemplate
+			);
+			$response = $apiClient->getSerializer()->deserialize($rawResponse, 'object', $httpHeader);
+		} catch (\Swagger\Client\ApiException $e) {
+			switch ($e->getCode()) {
+				case 204:
+					$data = $apiClient->getSerializer()->deserialize($e->getResponseBody(), 'object', $e->getResponseHeaders());
+					$e->setResponseObject($data);
+					break;
+				case 403:
+					$data = $apiClient->getSerializer()->deserialize($e->getResponseBody(), '\Swagger\Client\Discussion\Models\HalProblem', $e->getResponseHeaders());
+					$e->setResponseObject($data);
+					break;
+			}
+			throw $e;
+		}
+
+		if ($response['postCount'] == 0) {
+			// no posts
+			return false;
+		}
+
+		// extract post creation date from response
+		$mostRecentPostCreationDate = $response['_embedded']->{'doc:posts'}[0]->creationDate->epochSecond;
```

Comment:
This line is a little confusing; it uses three different notation types (square brackets, curly braces and arrow notation) for what I'm assuming are all stdClass values.  Maybe pick one?  I know 'doc:posts' would not be accessible as a stdClass member (e.g. `$response->_embedded->doc:posts`) due to the colon, so maybe standard square bracket array notation would be best.